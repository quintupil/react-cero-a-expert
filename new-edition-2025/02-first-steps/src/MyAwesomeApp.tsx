import type { CSSProperties } from "react";

const firstName = "Carlos";
const lastName = "GarcÃ­a";
const favoriteGames = ["Elden Ring", "Smash", "Metal Gear"];
const isActive = false;
const address = {
  zipCode: "ABC-123",
  country: "Chile",
};

const myStyle: CSSProperties = {
  backgroundColor: "#fafafa",
  borderRadius: 20,
  padding: 10,
  marginTop: 30,
};

export const MyAwesomeApp = () => {
  return (
    <>
      {/* Lo que va entre corchete son expresiones validas de JavaScript*/}
      <h1>{firstName}</h1>
      <h3>{lastName}</h3>
      <p>{favoriteGames.join(", ")}</p>
      <h1>{isActive ? "Activo" : "No activo"}</h1>
      <p style={myStyle}>{JSON.stringify(address)}</p>
    </>
  );
};
