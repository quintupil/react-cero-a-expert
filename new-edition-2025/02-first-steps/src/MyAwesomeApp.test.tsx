import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from "@testing-library/react";

describe("MyAwesomeApp", () => {
  test("should render firstName and lastName", () => {
    //console.log(document.body);

    /**Primera forma de utilizar render**/
    const { container } = render(<MyAwesomeApp />);
    //console.log(container);
    //console.log(container.innerHTML);

    /**Segunda forma de utilizar render**/
    //render(<MyAwesomeApp />);
    //console.log(screen.debug());
    screen.debug();
    const h1 = container.querySelector("h1");
    const h3 = container.querySelector("h3");
    //console.log(h1?.innerHTML);
    //expect(h1?.innerHTML).toBe("Fernando");
    expect(h1?.innerHTML).toContain("Carlos");
    expect(h3?.innerHTML).toContain("GarcÃ­a");
  });

  test("should render firstName and lastName - screen", () => {
    render(<MyAwesomeApp />);
    screen.debug();
    //const h1 = screen.getByRole("heading", {
    //level: 1,
    //});
    const h1 = screen.getByTestId("first-name-title");
    //console.log(h1.innerHTML);
    expect(h1.innerHTML).toContain("Carlos");
  });

  test("should match snapshot", () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot", () => {
    render(<MyAwesomeApp />);
    expect(screen.getByTestId("div-app")).toMatchSnapshot();
  });
});
