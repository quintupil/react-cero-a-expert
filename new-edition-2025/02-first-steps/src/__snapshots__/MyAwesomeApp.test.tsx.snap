// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Carlos
    </h1>
    <h3>
      García
    </h3>
    <p>
      4
    </p>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Smash, Metal Gear
    </p>
    <h1>
      No activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","country":"Chile"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Carlos
  </h1>
  <h3>
    García
  </h3>
  <p>
    4
  </p>
  <p
    class="mi-clase-favorita"
  >
    Elden Ring, Smash, Metal Gear
  </p>
  <h1>
    No activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC-123","country":"Chile"}
  </p>
</div>
`;
